src/python/pants/backend/python/subsystems/pytest.py

```diff
@@ -76,6 +76,13 @@ def register_options(cls, register):
             advanced=True,
             help="The format of the generated XML file. See https://docs.pytest.org/en/latest/reference.html#confval-junit_family.",
         )
+        register(
+            "--execution-slot-var",
+            type=str,
+            default="",
+            advanced=True,
+            help="If a non-empty string, the process execution slot id (an integer) will be exposed to tests under this environment variable name.",
```

Comment:
Would be a bit cleaner for this to default to `None` (`Optional[str]`) and in the `Process` constructor, even if inside of `Process` it gets converted into the empty string.

Reply:
Does the options `type` kwarg know how to interpret modern Python `typing` type annotations?

Reply:
No. You keep `type=str` and set `default=None`. The type will end up being `Optional[str]`.